source/extensions/dynamic_modules/sdk/rust/src/lib.rs

```diff
@@ -41,3 +58,111 @@ macro_rules! declare_program_init {
         }
     };
 }
+
+/// The function signature for the program init function.
+///
+/// This is called when the dynamic module is loaded, and it must return true on success, and false on failure. When it returns false,
+/// the dynamic module will not be loaded.
+///
+/// This is useful to perform any process-wide initialization that the dynamic module needs.
+pub type ProgramInitFunction = fn() -> bool;
+
+/// The function signature for the new HTTP filter configuration function.
+///
+/// This is called when a new HTTP filter configuration is created, and it must return a new instance of the [`HttpFilterConfig`] object.
+/// Returning `None` will cause the HTTP filter configuration to be rejected.
+//
+// TODO(@mathetake): I guess there would be a way to avoid the use of dyn in the first place.
+// E.g. one idea is to accept all concrete type parameters for HttpFilterConfig and HttpFilter traits in declare_init_functions!,
+// and generate the match statement based on that.
+pub type NewHttpFilterConfigFunction = fn(
+    envoy_filter_config: EnvoyHttpFilterConfig,
+    name: &str,
+    config: &str,
+) -> Option<Box<dyn HttpFilterConfig>>;
+
+/// The global init function for HTTP filter configurations. This is set via the `declare_init_functions` macro,
+/// and is not intended to be set directly.
+pub static mut NEW_HTTP_FILTER_CONFIG_FUNCTION: NewHttpFilterConfigFunction = |_, _, _| {
+    panic!("NEW_HTTP_FILTER_CONFIG_FUNCTION is not set");
+};
```

Comment:
This is linked/defined only in the .so right? So there will be no sharing issues?

Reply:
not really - each module is loaded with `RTLD_LOCAL` flag so any symbols won't collide with other dynamic modules:

https://github.com/envoyproxy/envoy/blob/090e73d82462f46d32c112554206a5ed0861eba2/source/extensions/dynamic_modules/dynamic_modules.cc#L21-L24